docs/Using Randomness/v2-adhoc/design.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: nodes.liquid
+section: ethereum
+date: Last Modified
+title: 'Ad-hoc Method'
+permalink: 'docs/vrf/v2/ad-hoc/'
+whatsnext:
+  {
+    'Get a Random Number': '/docs/vrf/v2/ad-hoc/get-a-random-number/',
+    'Configuration': '/docs/vrf/v2/ad-hoc/configuration/',
+  }
+metadata:
+  title: 'Generate Random Numbers for Smart Contracts using Chainlink VRF v2 - Ad-hoc method'
+  description: 'Learn how to securely generate random numbers for your smart contract with Chainlink VRF v2(an RNG). This guide uses the ad-hoc method.'
+---
+
+> ℹ️You are viewing the VRF v2 guide - Ad-hoc method.
+>
+> - To learn how to request random numbers with a subscription, see the [Subscription Method](/docs/vrf/v2/subscription/) guide.
+>
+> - If you are using v1, see the [VRF v1 guide](/docs/vrf/v1/introduction/).
+
+**Table of contents**
+
+- [Overview](#overview)
+- [Concepts](#concepts)
+  - [VRF Ad-hoc](#vrf-ad-hoc)
+  - [Request and Receive Data](#request-and-receive-data)
+    - [End To End Diagram](#end-to-end-diagram)
+    - [Explanation](#explanation)
+  - [Limits](#limits)
+
+## Overview
+
+This section explains how to generate random numbers using the ad-hoc method. This method doesn't require managing subscriptions and is more suited to one-off requests for randomness or situations where it is required to transfer the cost of VRF to end users as the cost is known at request time.
```

Comment:
```suggestion
This guide explains how to generate random numbers using the ad-hoc method. This method doesn't require a subscription and is optimal for one-off requests for randomness. This method also works best for applications where your end-users must pay the fees for VRF because the cost of the request is determined at request time.
```